torch/csrc/utils/invalid_arguments.cpp

```diff
@@ -272,7 +272,32 @@ std::string _formattedArgDesc(
       result += red;
     if (is_kwarg)
       result += option.arguments[i].name + "=";
-    result += py_typename(arg);
+    bool is_tuple = PyTuple_Check(arg);
+    if (is_tuple || PyList_Check(arg)) {
+      result += py_typename(arg) + " of ";
+      auto num_elements = PySequence_Length(arg);
+      if (is_tuple) {
+        result += "(";
+      } else {
+        result += "[";
+      }
+      for (const auto i : c10::irange(num_elements)) {
+        if (i != 0) {
+          result += ", ";
+        }
+        result += py_typename(py::reinterpret_steal<py::object>(PySequence_GetItem(arg, i)));
```

Comment:
OOPS good catch

Reply:
Yes PySequence is the only one that returns a fresh reference.